File: 724-find-pivot-index/724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        ls = 0
        rs = sum(nums)
        for i in range(len(nums)):
            ls += nums[i]
            if ls == rs:
                 return i
            rs -= nums[i]
        return -1

File: 724-find-pivot-index/724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int ls = 0;
        int rs = 0;
        for(int i = 0; i < nums.length; i++){
            rs += nums[i];
        }
        for(int i = 0; i< nums.length; i++){
            ls += nums[i];
            if(ls==rs){
                return i;
            }
            rs -= nums[i];
        }
        return -1;
    }
}